frontend/src/routes/BareMetalAssets/BareMetalAsset.test.tsx

```diff
@@ -139,46 +146,42 @@ describe('bare metal asset page', () => {
                 getDeleteBMAResourceAttributes('test-bare-metal-asset-001', 'test-bare-metal-asset-namespace')
             ),
         ]
-        const { getByText, getAllByText, getAllByLabelText, queryByText } = render(
-            <MemoryRouter>
-                <BareMetalAssetsPage />
-            </MemoryRouter>
+        render(
+            <RecoilRoot initializeState={(snapshot) => snapshot.set(bareMetalAssetsState, mockBareMetalAssets)}>
+                <MemoryRouter>
+                    <BareMetalAssetsPage />
+                </MemoryRouter>
+            </RecoilRoot>
         )
 
         await waitFor(() => expect(clusterNock.isDone()).toBeTruthy())
-        await waitFor(() => expect(listNock.isDone()).toBeTruthy()) // expect the list api call to finish
-        await waitFor(() => expect(getAllByText(mockBareMetalAssets[0].metadata.name!).length > 0)) // check for asset in doc
-        userEvent.click(getAllByLabelText('Actions')[0])
+        await waitForText(mockBareMetalAssets[0].metadata!.name!)
+        await clickByLabel('Actions', 0) // Click the action button on the first table row
         await waitFor(() => expect(nocksAreDone(rbacNocks)).toBeTruthy())
-        userEvent.click(getByText('bareMetalAsset.rowAction.deleteAsset.title')) // click the delete action
-        expect(getByText('common:delete')).toBeInTheDocument()
-        userEvent.click(getByText('common:delete')) // click confirm on the delete dialog
-        await waitFor(() => expect(deleteNock.isDone()).toBeTruthy()) // expect the delete api call to finish
-        expect(queryByText('test-bare-metal-asset-1')).toBeNull() // expect asset to no longer exist in doc
```

Comment:
I noticed that in the providerConnection code, assertions confirming the result of a delete were removed following the stream api change. I adopted that approach tentatively, to align my code with the changes that have already been established/PR'd. I do wonder if this is not a step backwards, as far as thorough testing goes.

Reply:
Also, I recognize that streaming does not seem to be totally compatible with our nock method. I would guess that's why this change was made in the first place.